contracts/root/RootERC20Predicate.sol

```diff
@@ -108,17 +114,36 @@ contract RootERC20Predicate is Initializable, IRootERC20Predicate {
         return childToken;
     }
 
-    function _deposit(IERC20Metadata rootToken, address receiver, uint256 amount) private {
-        address childToken = rootTokenToChildToken[address(rootToken)];
+    function _mapNative() private {
+        require(rootTokenToChildToken[address(0)] == address(0), "RootERC20Predicate: ALREADY_MAPPED");
 
-        if (childToken == address(0)) {
-            childToken = mapToken(rootToken);
-        }
+        address childPredicate = childERC20Predicate;
+
+        address childToken = Clones.predictDeterministicAddress(
+            childTokenTemplate,
+            keccak256(abi.encodePacked(address(0))),
```

Comment:
Use a constant for this address(0) and the subsequent one: they are the special address for the native token: Ether.